Source/WebCore/platform/graphics/gstreamer/GStreamerCommon.h

```diff
@@ -224,78 +224,81 @@ class GstMappedFrame {
     WTF_MAKE_NONCOPYABLE(GstMappedFrame);
 public:
 
-    GstMappedFrame(GstBuffer* buffer, GstVideoInfo* info, GstMapFlags flags)
+    static std::unique_ptr<GstMappedFrame> create(GstBuffer* buffer, GstVideoInfo* info, GstMapFlags flags)
     {
-        m_isValid = gst_video_frame_map(&m_frame, info, buffer, flags);
+        return create(new GstMappedFrame(buffer, info, flags));
```

Comment:
The point of having create methods is to remove the m_isValid, and make sure we can only create valid GstMappedFrame objects, returning nullptr from create in case of invalid without having to create a GstMappedFrame object. So, `gst_video_frame_map` should be called from the create, and the resulting valid GstVideoFrame passed to the constructor

Reply:
This is outdated, I think it is better to keep that in order to have it aligned with GstMappedBuffer.